platform/src/main/java/org/hillview/sketches/FindSketch.java

```diff
@@ -96,26 +98,26 @@ public JsonElement toJsonTree() {
      */
     private final RecordOrder recordOrder;
 
-    public FindSketch(final StringFilterDescription toFind,
+    public FindSketch(final StringFilterDescription stringFilterDescription,
                       final @Nullable RowSnapshot topRow,
                       final RecordOrder recordOrder) {
-        this.toFind = toFind;
+        this.stringFilterDescription = stringFilterDescription;
         this.topRow = topRow;
         this.recordOrder = recordOrder;
     }
 
     @Override
     public Result create(ITable data) {
         long count = 0;
-        IStringFilter filter = this.toFind.getFilter();
         IRowIterator rowIt = data.getRowIterator();
         Schema toCheck = this.recordOrder.toSchema();
+        IStringFilter stringFilter = new StringFilterFactory().getFilter(this.stringFilterDescription);
```

Comment:
In general factories you do not allocate; they implement a singleton pattern.
So a factory should have a static field called instance which is statically allocated.

Reply:
Perhaps then factory is the wrong term for this class? It needs to allocate a different type of filter based on the stringFilterDescription it receives. So I don't see how to do that statically.